crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -729,6 +732,22 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
                 else {
                     panic!("Unbound package '{package}' for upgrade");
                 };
+
+                let mut original_package_addrs = vec![];
+                for dep in dependencies.iter() {
+                    let named_address_mapping = &mut self.compiled_state.named_address_mapping;
+                    let dep = &Symbol::from(dep.as_str());
+                    let Some(orig_package) = self.package_upgrade_mapping.get(dep) else { continue };
+                    let Some(orig_package_address) = named_address_mapping.insert(orig_package.to_string(), zero) else { continue };
```

Comment:
In a [previous comment there was an idea](https://github.com/MystenLabs/sui/pull/11128#discussion_r1174342341) to remove the `orig_package` here entirely without setting to `zero`. After trying this, it turns out we can't remove it, the compiler still expects it in the mapping.

Reply:
In the [linkage test when we upgrade the last package](https://github.com/MystenLabs/sui/pull/11128/files#diff-08ed74f0dfa2e4a5f3086b6f99fc1051ea936a471e27d247da91de1b8457a17dR34-R41) I get:

```
Error: error[E03001]: address with no value
   ┌─ ...
   │
28 │     use Test_DepV1::DepM1;
   │         ^^^^^^^^^^ address 'Test_DepV1' is not assigned a value

error[E03001]: address with no value
  ┌─ ...
  │
7 │ module Test_DepV1::DepM1 {
  │        ^^^^^^^^^^ address 'Test_DepV1' is not assigned a value. Try assigning it a value when calling the compiler
```

> can't figure out where we would refer to the original name

So the unexpected reference to the original is coming from the package being upgraded, which I _think_ is coming from the fact that we are passing `Test_V2` to the compiler (I haven't looked deeply into this and will report back, but doesn't feel like a blocking issue).